src/App.tsx

```diff
@@ -15,61 +14,132 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
-}
-
-// Use this function in the render method
-function getReorderedGoods(
+type State = {
   goods: string[],
-  sortType: SortType,
+  isVisible: boolean,
   isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
+  sortBy: string,
+};
 
-  // Sort and reverse goods if needed
-  // ...
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    isVisible: true,
+    isReversed: false,
+    sortBy: '',
+  };
 
-  return visibleGoods;
-}
+  toShowGoods = () => {
+    this.setState({ isVisible: false });
+  };
 
-// DON'T save goods to the state
-type State = {
-  isStarted: boolean,
-  isReversed: boolean,
-  sortType: SortType,
-};
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState({ sortBy: 'alphabet' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
+  reset = () => {
+    this.setState({ isReversed: false, sortBy: '' });
+  };
 
-    <button type="button">
-      Sort alphabetically
-    </button>
+  render() {
+    const {
+      goods,
+      isReversed,
+      sortBy,
+      isVisible,
+    } = this.state;
 
-    <button type="button">
-      Sort by length
-    </button>
+    const copyGoods = [...goods];
 
-    <button type="button">
-      Reverse
-    </button>
+    copyGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case 'length':
+          return good1.length - good2.length;
+        case 'alphabet':
+          return good1.localeCompare(good2);
 
-    <button type="button">
-      Reset
-    </button>
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      copyGoods.reverse();
+    }
```

Comment:
This logic should be moved to a separate method. `render` should be as clear as possible and without any extra logic

Reply:
done